.github/workflows/build.yml

```diff
@@ -3,9 +3,34 @@ name: Pipeline
 on: [push, pull_request]
 
 jobs:
-  build:
-    name: "Build, Test and Publish"
-    runs-on: ubuntu-latest
+  test-linux:
+    uses: ./.github/workflows/test-target.yml
+    with:
+      runs-on: ubuntu-latest
+      tasks: check # all linux targets + JVM + JS + WasmJS
+  test-windows:
+    uses: ./.github/workflows/test-target.yml
+    with:
+      runs-on: windows-latest
+      tasks: mingwX64Test
+  test-macosX64:
+    uses: ./.github/workflows/test-target.yml
+    with:
+      runs-on: macos-13 # latest version with X64 processor
+      tasks: macosX64Test iosX64Test
+  test-macosArm64:
+    uses: ./.github/workflows/test-target.yml
+    with:
+      runs-on: macos-latest
+      tasks: macosArm64Test iosSimulatorArm64Test
```

Comment:
Couldn't this be a single job with a [matrix](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrixinclude)? We could just use `jobs.*.strategy.matrix.include` to explicitly define the jobs and the tasks we want them to run.

Reply:
Yes, it could be. But it won't be possible to use a reusable workflow in matrix job (makes it a bit less readable in my opinion).
I will convert those jobs to a matrix job and you can compare both options